tests/test_scripts.py

```diff
@@ -1,16 +1,14 @@
-import os
 import subprocess
+from pathlib import Path
 
 import pytest
 
 TEST_NAMES = [
-    name[5:-3]
-    for name in os.listdir("tests/scripts")
-    if name.startswith("test_") and name.endswith(".sh")
+    str(name)[14:-3].replace("/", ".") for name in Path("tests/scripts").glob("**/test_*.sh")
```

Comment:
This will find any test script file in any nested subdirectory, right? This does not conform to how you do the relative lookup of the config in `config.source`.
What if we fold the tests inside out: Have a setup script to provide the needed functions and env-vars and then source a specific test. It would also remove some repeated weirdness from the tests. It may make it impossible to run the tests by simply calling that script however, but with `pytest -k` we should be ok there.

Is it problematic to have "/" in the test name? I'd rather not replace all "/" with dots an than all dots with "/", as it's not a noop.

Reply:
> This will find any test script file in any nested subdirectory, right? This does not conform to how you do the relative lookup of the config in config.source.

How so? 

> Is it problematic to have "/" in the test name? I'd rather not replace all "/" with dots an than all dots with "/", as it's not a noop.

It is indeed a problem:

```
$ pytest -k "test/test"
ERROR: Wrong expression passed to '-k': test/test_remote: at column 5: unexpected character "/"
```

Reply:
I see what you mean about config.source. I can look at the pattern you suggest of inverting config.source. Perhaps I could rename it as test_runner.sh and have it accept a list of `TESTS`?